src/pay.rs

```diff
@@ -315,13 +315,15 @@ impl Runtime {
         let contract_id = invoice.contract.ok_or(CompletionError::NoContract)?;
 
         let fascia = psbt.rgb_commit()?;
-        if let Some(output) = psbt.dbc_output::<TapretProof>() {
-            let terminal = output
-                .terminal_derivation()
-                .ok_or(CompletionError::InconclusiveDerivation)?;
-            let tapret_commitment = output.tapret_commitment()?;
-            self.wallet_mut()
-                .add_tapret_tweak(terminal, tapret_commitment)?;
+        if let (Some(_), _) = fascia.anchor.as_reduced_unsafe().as_split() {
```

Comment:
This is dangerous way to do it - if the `split` method change signature and will return opret anchor first, this will be missed to be changed. Thus, the better is to
```suggestion
        if let (Some(Anchor::<_, TapretProof> { .. }), _) = fascia.anchor.as_reduced_unsafe().as_split() {
```

... but this is too clunky and better I would modify RGB Core API